### ISSUE TITLE
Finetune nnUNet for different number of modality/ channels than pre-trained one

### ISSUE THREAD

**Issue by mehnaz1985** [MIC-DKFZ/nnUNet#2252]
I pretrained nnUNet on the BraTs2020 dataset, where the input channel is 4 (t1, t2, t1ce, & flair) with 3 lebels (ED/NET, TC, and ET), and now I would like to fine-tune that for one private dataset that has 7 channels (t1, t2, t1ce, flair, dti_ra, dti_fa, & dti_md) with 2 lebels. I followed 
https://github.com/MIC-DKFZ/nnUNet/issues/774
 
and could manage to work for 4 channels with 2 lebels, but not with 7 channels. Can anyone help me how I can feed 7 channels in that using the pre-trained weight? I tried that (7 channels with lebel 2) from scratch but it didn't give me any acceptable results.

**Comment by Lars-Kraemer** [comment]
Hey @mehnaz1985 

This is not supported by -pretrained_weights as-is. In nnU-Net v2, pretrained weights are only loaded if parameter names and shapes match; only the segmentation layers are excluded automatically. So changing from 3 labels to 2 labels is okay, but changing the input from 4 channels to 7 channels changes the first convolution shape and makes the checkpoint incompatible. If you want to use the BraTS model, you would need to implement custom weight loading for the first layer (for example copy the pretrained weights for the 4 MRI channels and initialize the 3 DTI channels separately). The default nnU-Net loader cannot do that automatically

Best,
Lars
